sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -213,6 +223,14 @@ class SpanData final : public Recordable
     span_kind_ = span_kind;
   }
 
+  void SetResourceAttribute(
```

Comment:
I'm trying to understand what this method is intended to do and where it is in the spec.

Fro what I gather, this is basically copying all the resource-attributes onto the span.  Can we instead just grab a reference to the resource?